Source/Core/Core/IOS/USB/Emulated/Skylanders/SkylanderFigure.cpp

```diff
@@ -0,0 +1,449 @@
+        decrypted[area_offset + 0x9]++;
+      else
+      {
+        decrypted[area_offset + 0x9] += 2;
+      }
+      ComputeChecksum(ChecksumType::Type1, decrypted.data() + area_offset,
+                      decrypted.data() + area_offset + 0xE);
+    }
+
+    {
+      area_offset = (decrypted[0x112] != (decrypted[0x2D2] + 1U)) ? 0x110 : 0x2D0;
+
+      if (decrypted[0x112] == decrypted[0x2D2])
+        decrypted[area_offset + 0x2]++;
+      else
+        decrypted[area_offset + 0x2] += 2;
+
+      ComputeChecksum(ChecksumType::Type6, decrypted.data() + area_offset,
+                      decrypted.data() + area_offset);
+
+      area_offset = (area_offset == 0x110) ? 0x2D0 : 0x110;
+
+      ComputeChecksum(ChecksumType::Type6, decrypted.data() + area_offset,
+                      decrypted.data() + area_offset);
+    }
+  }
+  else if (figure_data->normalized_type == Type::Trophy)
+  {
+    {
+      u16 area_offset = (decrypted[0x89] != (decrypted[0x249] + 1U)) ? 0x80 : 0x240;
+
+      memcpy(decrypted.data() + area_offset + 0x14, &figure_data->trophy_data.unlocked_villains, 1);
+
+      {
+        ComputeChecksum(ChecksumType::Type3, decrypted.data() + area_offset + 0x50,
+                        decrypted.data() + area_offset + 0xA);
+        ComputeChecksum(ChecksumType::Type2, decrypted.data() + area_offset + 0x10,
+                        decrypted.data() + area_offset + 0xC);
+        if (decrypted[0x89] == decrypted[0x249])
+          decrypted[area_offset + 0x9]++;
+        else
+          decrypted[area_offset + 0x9] += 2;
+        ComputeChecksum(ChecksumType::Type1, decrypted.data() + area_offset,
+                        decrypted.data() + area_offset + 0xE);
+      }
+
+      area_offset = (area_offset == 0x80) ? 0x240 : 0x80;
+      {
+        ComputeChecksum(ChecksumType::Type3, decrypted.data() + area_offset + 0x50,
+                        decrypted.data() + area_offset + 0xA);
+        ComputeChecksum(ChecksumType::Type2, decrypted.data() + area_offset + 0x10,
+                        decrypted.data() + area_offset + 0xC);
+        ComputeChecksum(ChecksumType::Type1, decrypted.data() + area_offset,
+                        decrypted.data() + area_offset + 0xE);
+      }
+    }
+
+    {
+      u16 area_offset = (decrypted[0x112] != (decrypted[0x2D2] + 1U)) ? 0x110 : 0x2D0;
+
+      if (decrypted[0x112] == decrypted[0x2D2])
+        decrypted[area_offset + 0x2]++;
+      else
+        decrypted[area_offset + 0x2] += 2;
+
+      ComputeChecksum(ChecksumType::Type6, decrypted.data() + area_offset,
+                      decrypted.data() + area_offset);
+
+      area_offset = (area_offset == 0x110) ? 0x2D0 : 0x110;
+
+      ComputeChecksum(ChecksumType::Type6, decrypted.data() + area_offset,
+                      decrypted.data() + area_offset);
+    }
+  }
+
+  // Encrypt again
+  Encrypt(decrypted);
+
+  Save();
+}
+void SkylanderFigure::DecryptFigure(u8 dest[FIGURE_SIZE]) const
```

Comment:
`u8 dest[FIGURE_SIZE]` -> `u8* dest` so it matches the header. Or just make it `std::array<u8, FIGURE_SIZE>* dest`, since it only ever seems to be called with that anyway.